enhancements/audit-command.md

```diff
@@ -0,0 +1,642 @@
+
+See that we might want to improve the presentation of this information in a follow up of this proposal.
+
+**Changes in the catalog image format**
+
+The proposal [Package representation and management in an index](https://github.com/operator-framework/enhancements/blob/master/enhancements/declarative-index-config.md) will change the format of the index. Then, if the audit command to be implemented before the change it would mean that would be required to replace the SQL queries code implementations to manipulate the JSON instead.
+
+**SDK using OPM binary**
+
+If the audit command be shipped in the SDK then, we also would have the same current risks described in the proposal [operator sdk builds catalogs](https://github.com/operator-framework/enhancements/blob/master/enhancements/operator-sdk-builds-catalogs.md#risks-and-mitigations) regarding the dependency with OLM binary only if we decide to use it to extract the operator bundle from the index catalog. The alternative solution would be re-implement this feature in SDK as well which is reasonable since it will require changes to address the audit needs.
+
+## Design Details
+
+### Test Plan
+
+For the tests purpose we will create a mock index catalog image with only a few operator bundles to cover the main scenarios and use it in the e2e tests. The idea is to ensure the command without make this test take to long to be executed. 
+
+### Version Skew Strategy
+
+The audit command does not require support and work with the previous index versions or the old Package Manifest format.
+
+## Drawbacks
+
+To generate the full report we might need to create, for example, a CronJob process since it can take too long to be executed.
+
+The time to process the full report is proportional to the quantity of operator bundles be in the index catalog image and of projects which are not built with SDK and its latest version(s) since will be required to more requests against the project repositories to gathering the information.
+
+However, see the [POC](
+https://github.com/jmccormick2001/indexdump) project which proves that this command operation is achievable.  
+
+
+## Implementation History
+
+Note that this proposal shows replace the need of others open pull requests such as:
+- [Enhancement for validating bundles in a catsrc image #48](https://github.com/operator-framework/enhancements/pull/48). The audit command shows cover all requirements raised in this proposal.
+- [Custom operator validation #43](https://github.com/operator-framework/enhancements/pull/43): The audit command will able to use scorecard which let the operator bundle developers create their own specific tests for their bundles. In this way, the requirement to allow extending the Validators from the [operator-framework/api](https://github.com/operator-framework/api) might no longer be need and or to be a priority.
+
+[ep-new-checks for-operator-hub]: https://github.com/operator-framework/enhancements/pull/65
```

Comment:
how does this EP replace the need for #65 ?  it seems to build on top of #65

Reply:
No. It is a link alias on the bottom. The EP replaces the above 2 proposals only.

Reply:
got it.